### PR TITLE
[Projects] Stop logs for runs when deleting project

### PR DESCRIPTION
**Pull request by TomerShor**
A followup PR to #3082.

When deleting a project, call `stop_logs` for all of the project's runs before they and their resources(pods) are deleted.
This is done so the log collector won't get stuck trying to collect logs for non-existing runs.

Since the log collector client is async, this required moving all of the `delete_project` functions from sync to async.

### CODE REVIEW

File: mlrun/api/crud/projects.py [hunk 1]
```diff
@@ -72,7 +73,7 @@ def patch_project(
             session, name, project, patch_mode
         )
 
-    def delete_project(
+    async def delete_project(
```
Comment (liranbg) [inline mlrun/api/crud/projects.py:76]: once you graduate a function to be async, you *must* follow all function inside calls to make sure all calls to IO are either async + await or sync and runs in threadpools. e.g. `verify_project_has_no_related_resources` must be run in threadpool.

File: mlrun/api/utils/projects/follower.py [hunk 2]
```diff
@@ -45,7 +46,7 @@ class Member(
     mlrun.api.utils.projects.member.Member,
     metaclass=mlrun.utils.singleton.AbstractSingleton,
 ):
-    def initialize(self):
+    async def initialize(self):
```
Comment (liranbg) [inline mlrun/api/utils/projects/follower.py:49]: see comment above

File: mlrun/api/utils/projects/follower.py [hunk 3]
```diff
@@ -315,10 +316,12 @@ def _sync_projects(self, full_sync=False):
 
         db_session = mlrun.api.db.session.create_session()
```
Comment (liranbg) [inline mlrun/api/utils/projects/follower.py:317]: ⚠️ blockig io in async

**Review by liranbg — CHANGES_REQUESTED** [review]
this is a very dangerous change late in 1.3.0 development. lets discuss offline to see how to make such change with zero risk.

**Comment by TomerShor** [comment]
Implemented in https://github.com/mlrun/mlrun/pull/3145

**Pull request closed without merging**